lib/figgy/configuration.rb

```diff
@@ -40,14 +38,45 @@ def initialize
       define_handler 'json' do |contents|
         JSON.parse(contents)
       end
+
+      @overlays = [Overlay.new('root', nil, @roots)]
     end
 
+    # Sets the +root+ to the given directory, or to a Vault client.
+    #
+    # If just a path is given, it sets the root to that file directory
+    # path. If a path and Vault client are given, it sets the root to
+    # a Vault client rooted at the given path.
+    def set_root(path, vault = nil)
+      new_root = if vault
+                   VaultRoot.new(vault, path)
+                 else
+                   FileRoot.new(File, File.expand_path(path))
+                 end
+
+      @roots = [new_root]
+      @overlays = [Overlay.new('root', nil, @roots)]
+    end
+
+    # Sets the +root+ to the given directory.
+    #
+    # For use with file directory roots only.
+    #
+    # @see #set_root
     def root=(path)
-      @roots = [File.expand_path(path)]
+      @roots = [FileRoot.new(File, File.expand_path(path))]
+      @overlays = [Overlay.new('root', nil, @roots)]
     end
```

Comment:
Not sure how we feel about breaking changes, is this worth keeping?

Reply:
Do you mean keeping the `#root=` method as opposed to replacing it with two new methods `#set_vault_root` + `#set_file_root`? If so, I'd definitely rather keep `#root=` right now, as most figgy configs are probably set up with it.